packages/labs/virtualizer/src/test/scenarios/element-and-directive-parity.test.ts

```diff
@@ -96,45 +129,67 @@ describe('lit-virtualizer and virtualize directive', () => {
     const ulv: UsingLitVirtualizer = example.querySelector(
       'using-lit-virtualizer'
     )!;
+    const uvd: UsingVirtualizeDirective = example.querySelector(
+      'using-virtualize-directive'
+    )!;
+
+    ulv.items = [...items];
+    uvd.items = [...items];
 
-    ulv.items = items;
     ulv.selected = selected;
+    uvd.selected = selected;
 
-    await until(() => ulv.shadowRoot?.textContent?.includes('2 selected'));
+    await until(() => ulv.shadowRoot?.textContent?.includes('[5 selected]'));
+    await until(() => uvd.shadowRoot?.textContent?.includes('[5 selected]'));
 
-    expect(ulv.shadowRoot?.textContent).to.include('2 selected');
-    expect(ulv.shadowRoot?.textContent).to.include('5 selected');
+    expect(ulv.shadowRoot?.textContent).to.include('[2 selected]');
+    expect(uvd.shadowRoot?.textContent).to.include('[2 selected]');
 
-    const uvd: UsingVirtualizeDirective = example.querySelector(
-      'using-virtualize-directive'
-    )!;
+    expect(ulv.shadowRoot?.textContent).to.include('[5 selected]');
+    expect(uvd.shadowRoot?.textContent).to.include('[5 selected]');
 
-    uvd.items = items;
-    uvd.selected = selected;
+    // Changing selection doesn't trigger visibility changed or range changed events.
+    ulv.selected = new Set([1, 3]);
+    uvd.selected = new Set([1, 3]);
+
+    await until(() => ulv.shadowRoot?.textContent?.includes('[3 selected]'));
+    await until(() => uvd.shadowRoot?.textContent?.includes('[3 selected]'));
+
+    expect(ulv.shadowRoot?.textContent).to.include('[1 selected]');
+    expect(uvd.shadowRoot?.textContent).to.include('[1 selected]');
 
-    await until(() => uvd.shadowRoot?.textContent?.includes('2 selected'));
+    expect(ulv.shadowRoot?.textContent).to.include('[3 selected]');
+    expect(uvd.shadowRoot?.textContent).to.include('[3 selected]');
 
-    expect(uvd.shadowRoot?.textContent).to.include('2 selected');
-    expect(uvd.shadowRoot?.textContent).to.include('5 selected');
+    expect(ulv.shadowRoot?.textContent).not.to.include('[2 selected]');
```

Comment:
Maybe not now, but since the test is so long, it'd be great to see some comments break it up and describe the scenario, what the setup is doing, what the interaction is, expectations are checking, etc.

Reply:
Fair points.  I'll add this when I do some overall test sprucing up in #2788